packages/ember-data/lib/serializers/json-serializer.js

```diff
@@ -749,7 +759,12 @@ export default Serializer.extend({
     @param {String} key
     @return {String} key
   */
-  _getMappedKey: function(key) {
+  _getMappedKey: function(key, modelClass) {
+    const hasAttributeKey = get(modelClass, 'attributes').has(key);
+    const hasRelationshipKey = get(modelClass, 'relationshipsByName').has(key);
+
+    Ember.assert('There is no attribute or relationship with the name `' + key + '` on `' + modelClass.modelName + '`. Check your serializers attrs hash.', hasAttributeKey || hasRelationshipKey);
```

Comment:
I don't think its possible for this assertion to get triggered since it looks like `_getMappedKey` always gets called in a `eachAttribute` or `eachRelationship` loop.